src/components/appointments/AppointmentStep.js

```diff
@@ -0,0 +1,147 @@
+import {Formik, useField} from 'formik';
+import React, {useContext} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+import {useAsync} from 'react-use';
+
+import {ConfigContext} from 'Context';
+import {get} from 'api';
+import Body from 'components/Body';
+import Button from 'components/Button';
+import Card from 'components/Card';
+import Label from 'components/Label';
+import Loader from 'components/Loader';
+import {Toolbar, ToolbarList} from 'components/Toolbar';
+import ValidationErrors from 'components/ValidationErrors';
+import {getBEMClassName, getFormattedDateString, getFormattedTimeString} from 'utils';
+
+const requestOrEmpty = async (url, query, condition) => {
+  if (condition) {
+    return await get(url, query);
+  } else {
+    return [];
+  }
+};
+
+const AppointmentField = ({
+  label,
+  getOptionValue,
+  getOptionLabel,
+  url,
+  urlQuery = {},
+  ...props
+}) => {
+  const [field, meta] = useField(props);
+  // load options from appointments api
+  const {loading, value: items} = useAsync(async () => {
+    if (urlQuery) {
+      const notEmpty = Object.values(urlQuery).every(val => !!val);
+      return await requestOrEmpty(url, urlQuery, notEmpty);
+    } else {
+      return await get(url);
+    }
+  }, Object.values(urlQuery));
```

Comment:
Objects in Javascript don't have inherent ordering, so relying on `Object.values` is not stable. The exact query params may still be the same, but the order shifted, which will re-trigger network requests.

I also find this makes the code hard to read in terms of which hook dependencies are specified. Maybe an Array of key-value pairs is a better datastructure?